src/main/java/uk/gov/hmcts/ccd/domain/service/lau/AuditCaseRemoteOperation.java

```diff
@@ -0,0 +1,172 @@
+package uk.gov.hmcts.ccd.domain.service.lau;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.AuditCaseRemoteConfiguration;
+import uk.gov.hmcts.ccd.auditlog.AuditEntry;
+import uk.gov.hmcts.ccd.auditlog.AuditOperationType;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.ccd.domain.model.lau.ActionLog;
+import uk.gov.hmcts.ccd.domain.model.lau.CaseActionPostRequest;
+import uk.gov.hmcts.ccd.domain.model.lau.CaseSearchPostRequest;
+import uk.gov.hmcts.ccd.domain.model.lau.SearchLog;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+@Service
+@Slf4j
+@Qualifier("AuditCaseRemoteOperation")
+public class AuditCaseRemoteOperation implements AuditRemoteOperation {
+
+    private static final String LAU_CASE_ACTION_CREATE = "CREATE";
+    private static final String LAU_CASE_ACTION_VIEW = "VIEW";
+    private static final String LAU_CASE_ACTION_UPDATE = "UPDATE";
+
+    public static final Map<String, String> CASE_ACTION_MAP = Map.of(
+        AuditOperationType.CREATE_CASE.getLabel(),LAU_CASE_ACTION_CREATE,
+        AuditOperationType.CASE_ACCESSED.getLabel(), LAU_CASE_ACTION_VIEW,
+        AuditOperationType.UPDATE_CASE.getLabel(), LAU_CASE_ACTION_UPDATE);
+
+    private final SecurityUtils securityUtils;
+
+    private final AuditCaseRemoteConfiguration auditCaseRemoteConfiguration;
+
+    private final HttpClient httpClient;
+
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    public AuditCaseRemoteOperation(@Lazy final SecurityUtils securityUtils,
+                                    @Qualifier("httpClientAudit") final HttpClient httpClient,
+                                    @Qualifier("SimpleObjectMapper") final ObjectMapper objectMapper,
+                                    final AuditCaseRemoteConfiguration auditCaseRemoteConfiguration) {
+        this.securityUtils = securityUtils;
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+        this.auditCaseRemoteConfiguration = auditCaseRemoteConfiguration;
+        objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
```

Comment:
```suggestion
```
Possible to delete this, or is it needed? If needed then safer to create a new one `ObjectMapper` directly in this class to avoid potential impact elsewhere

Reply:
Hi @danlysiak thanks for this. I've removed the objectMapper.setVisibility line and added lombok @Getters annotations to the new audit activity and search pojos.

Reply:
Notice build failed earlier in the day because Elastic Search was disabled?